src/shared/components/Tabs/Tabs.tsx

```diff
@@ -1,29 +1,112 @@
-import React, { useState } from 'react'
-import { TabsGroup, Tab } from './Tabs.styles'
+import { transitions } from '@/shared/theme'
+import { throttle } from 'lodash'
+import React, { useEffect, useRef, useState } from 'react'
+import { CSSTransition } from 'react-transition-group'
+import useResizeObserver from 'use-resize-observer'
+import { BackgroundGradient, Tab, TabsGroup, TabsWrapper, TAB_WIDTH } from './Tabs.styles'
 
-type TabsProps = {
+export type TabsProps = {
   tabs: string[]
   initialIndex?: number
   onSelectTab: (idx: number) => void
 }
 const Tabs: React.FC<TabsProps> = ({ tabs, onSelectTab, initialIndex = -1 }) => {
   const [selected, setSelected] = useState(initialIndex)
+  const [center, setCenter] = useState(0)
+  const tabsRef = useRef<HTMLDivElement>(null)
+  const [shadowsVisible, setShadowsVisible] = useState({
+    left: false,
+    right: true,
+  })
+
+  useResizeObserver<HTMLDivElement>({
+    ref: tabsRef,
+    onResize: ({ width }) => {
+      if (!tabsRef.current || !width) {
+        return
+      }
+      const isScrollBarVisible = tabsRef.current.scrollWidth > width
+      if (isScrollBarVisible) {
+        setCenter(width / 3)
+      } else {
+        setCenter(0)
+      }
+    },
+  })
+
+  useEffect(() => {
+    const tabsGroup = tabsRef.current
+    if (!tabsGroup) {
+      return
+    }
+    const currentItemOffsetleft = selected > 0 && TAB_WIDTH * selected
```

Comment:
I think we could simplify the flow here if we move the `selected > 0` condition to the `if` above and return if it's `< 1`